src/classes/OPP_StageMappingUtil.cls

```diff
@@ -93,8 +96,118 @@ public with sharing class OPP_StageMappingUtil {
     @testVisible private static void initializeStates() {
         // These are temporary values to be used only during interim for internal development on Abacus project.
         // This will be updated in March 2019 timeframe with revised code to retrieve from Custom Metadata instead
-        COMMITTED_STATE_SET = new Set<String> {'closed won'};
-        FINALIZED_STATE_SET = new Set<String> {'finalized'};
+        // COMMITTED_STATE_SET = new Set<String> {'closed won'};
+        // FINALIZED_STATE_SET = new Set<String> {'finalized'};
+
+        // If either the committed state set or the finalized state set is null, query for everything and 
+        // reinitialize everything. 
+        if (COMMITTED_STATE_SET == null || FINALIZED_STATE_SET == null) {
+            COMMITTED_STATE_SET = new Set<String>();
+            FINALIZED_STATE_SET = new Set<String>();
+
+            List<Opportunity_Stage_To_State_Mapping__mdt> stageToStateRecords = [Select Id, Opportunity_Stage__c, Opportunity_State__c FROM Opportunity_Stage_To_State_Mapping__mdt];
+
+            for (Opportunity_Stage_To_State_Mapping__mdt mappingRecord : stageToStateRecords) {
+                String oppStage = mappingRecord.Opportunity_Stage__c.trim().toLowerCase();
+
+                if (mappingRecord.Opportunity_State__c.trim().toLowerCase() == 'committed') {
+                    COMMITTED_STATE_SET.add(oppStage);
+                } else {
+                    FINALIZED_STATE_SET.add(oppStage);
```

Comment:
Technically, you could just use the map, and never worry about the sets, but you would have to modify the check to handle when no values are present, because the set would not be populated then.  Your call, though.   I am fine either way.  

The main reason I like to do this kind of thing this way, is that it reduces moving parts (meaning conditional branching), which is harder to write tests for.